src/mo_frontend/printers.ml

```diff
@@ -236,6 +236,252 @@ let string_of_symbol =
+  | X (N N_exp_nullary_bl_) -> exp
+  | X (N N_exp_nullary_ob_) -> exp
+  | X (N N_exp_plain) -> "true"
+  | X (N N_exp_post_bl_) -> exp
+  | X (N N_exp_post_ob_) -> exp
+  | X (N N_obj_or_class_dec) -> String.concat " " ["object"; pat; "= {}"]
+  | X (N N_option_exp_post_ob__) -> exp
+  | X (N N_exp_un_bl_) -> exp
+  | X (N N_exp_un_ob_) -> exp
+  | X (N N_func_body) -> "{}"
+  | X (N N_imp) -> imp
+  | X (N N_import_list) -> imp
+  | X (N N_inst) -> "<inst>"
+  | X (N N_lit) -> "true"
+  | X (N N_ob) -> exp_obj
+  | X (N N_obj_body) -> "{}"
+  | X (N N_option_EQ_) -> "=?"
+  | X (N N_option_exp_nullary_ob__) -> exp
+  | X (N N_option_typ_args_) -> typ_args
+  | X (N N_option_query_) -> "query"
+  | X (N N_parse_module_header) -> "<parse_module_header>"
+  | X (N N_parse_prog) -> "<parse_prog>"
+  | X (N N_parse_prog_interactive) -> "<parse_prog_interactive>"
+  | X (N N_parse_stab_sig) -> "<parse_stab_sig>"
+  | X (N N_pat) -> pat
+  | X (N N_pat_bin) -> pat
+  | X (N N_pat_field) -> pat ^ " " ^ annot
+  | X (N N_pat_nullary) -> pat
+  | X (N N_pat_plain) -> pat
+  | X (N N_pat_un) -> pat
+  | X (N N_path) -> "A.B.C"
+  | X (N N_annot_opt) -> annot
+  | X (N N_seplist_case_semicolon_) -> case ^ ";"
+  | X (N N_seplist_dec_SEMICOLON_) -> dec ^ ";"
+  | X (N N_seplist_dec_semicolon_) -> dec ^ ";"
+  | X (N N_seplist_typ_dec_semicolon_) -> typ_dec ^ ";"
+  | X (N N_seplist_dec_field_semicolon_) -> dec_field ^ ";"
+  | X (N N_seplist_exp_ob__COMMA_) -> exp
+  | X (N N_seplist_exp_field_semicolon_) -> exp_field ^ ";"
+  | X (N N_seplist1_exp_field_semicolon_) -> exp_field ^ ";"
+  | X (N N_separated_nonempty_list_AND_exp_post_ob__) -> exp
+  | X (N N_seplist_exp_nonvar_ob__COMMA_) -> exp
+  | X (N N_seplist_imp_SEMICOLON_) -> imp ^ ";"
+  | X (N N_seplist_imp_semicolon_) -> imp ^ ";"
+  | X (N N_seplist_pat_bin_COMMA_) -> pat
+  | X (N N_seplist_pat_field_semicolon_) -> pat ^ ";"
+  | X (N N_seplist_typ_COMMA_) -> typ
+  | X (N N_list_preceded_COMMA_typ__) -> ", " ^ typ
+  | X (N N_seplist_typ_bind_COMMA_) -> upat
+  | X (N N_list_preceded_COMMA_typ_bind__) -> ", " ^ upat
+  | X (N N_seplist_typ_field_semicolon_) -> typ_dec ^ ";"
+  | X (N N_seplist_stab_field_semicolon_) -> stab_field ^ ";"
+  | X (N N_seplist_typ_item_COMMA_) -> typ
+  | X (N N_seplist_typ_tag_semicolon_) -> typ_tag ^ ";"
+  | X (N N_seplist1_typ_tag_semicolon_) -> typ_tag ^ ";"
+  | X (N N_pat_opt) -> pat
+  | X (N N_typ) -> typ
+  | X (N N_typ_dec) -> typ_dec
+  | X (N N_typ_args) -> typ_args
+  | X (N N_typ_bind) -> upat
+  | X (N N_typ_field) -> typ_dec
+  | X (N N_typ_item) -> typ
+  | X (N N_typ_nobin) -> typ
+  | X (N N_typ_nullary) -> typ
+  | X (N N_typ_obj) -> "{}"
+  | X (N N_typ_pre) -> typ
+  | X (N N_typ_tag) -> typ_tag
+  | X (N N_typ_un) -> typ
+  | X (N N_typ_variant) -> "{ " ^ typ_tag ^ " }"
+  | X (N N_vis) -> "public"
+  | X (N N_stab) -> ""
+  | X (N N_stab_field) -> stab_field
+  | X (N N_start) -> "<start>" (* dummy non-terminal, don't display *)
```

Comment:
By "don't display", do you mean that the compiler never displays it, or that we should not display it?

Reply:
The same as above, comment comes from the `string_of_symbol` function.